sys/crypto/modes/ocb.c

```diff
@@ -320,7 +320,7 @@ int32_t cipher_decrypt_ocb(cipher_t *cipher, uint8_t *auth_data,
                            uint8_t *input, size_t input_len, uint8_t *output)
 {
 
-    if (input_len - tag_len > INT32_MAX) {
+    if (input_len > (uint32_t)(INT32_MAX - tag_len)) {
```

Comment:
This is functionally not equal to what is was before. was the implementation before incorrect compared to the spec?

Reply:
You are right. It should be fixed now.